test/index.spec.js

```diff
@@ -120,12 +120,14 @@ describe('Testing index', () => {
         }]
       })).to.equal(true);
       expect(await index.rest.data.refresh('offer')).to.equal(true);
-      expect(await index.rest.data.query('offer', index.query.build('offer', {
+      const filter = index.query.build('offer', {
         toReturn: ['id', 'meta'],
         filterBy: { and: [['id', '==', offerId]] },
         limit: 1,
         offset: 0
-      }))).to.deep.equal({
+      });
+      expect(index.data
+        .page(await index.rest.data.query('offer', filter), filter)).to.deep.equal({
```

Comment:
I wonder if we should separate out `queryResult = await index.rest.data.query(...);`? Would make this more readable which I like.